python/pyspark/mllib/linalg.py

```diff
@@ -897,6 +914,33 @@ def __init__(self, numRows, numCols, colPtrs, rowIndices, values,
             raise ValueError("Expected rowIndices of length %d, got %d."
                              % (self.rowIndices.size, self.values.size))
 
+    def __str__(self):
+        spstr = "{0} X {1} ".format(self.numRows, self.numCols)
+        if self.isTransposed:
+            spstr += "CSRMatrix\n"
+        else:
+            spstr += "CSCMatrix\n"
+
+        for i, colPtr in enumerate(self.colPtrs[:-1]):
+            endptr = self.colPtrs[i + 1]
+            values = self.values[colPtr: endptr]
+            rowindices = self.rowIndices[colPtr: endptr]
+            for j, rowInd in enumerate(rowindices):
+                if self.isTransposed:
+                    spstr += '({0},{1}) {2}\n'.format(
+                        i, rowInd, _format_float(values[j]))
+                else:
+                    spstr += '({0},{1}) {2}\n'.format(
+                        rowInd, i, _format_float(values[j]))
```

Comment:
How about this:
cur_col = 0
rs = []
for i, (rowIndice, value) in enumerate(izip(self.rowIndices, self.values)):
     while self.colPtrs[cur_col+1] <= i:
          cur_col += 1
     rs.append("(%s,%s) %s" % (cur_col, rowIndice, value))
return spstr +  '\n'.join(rs)